specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2025-03-03/gallery.json

```diff
@@ -2966,6 +3607,248 @@
+        "publishingProfile"
+      ],
+      "description": "Describes the properties of a gallery script version."
+    },
+    "GalleryScriptVersionPublishingProfile": {
+      "type": "object",
+      "properties": {
+        "source": {
+          "$ref": "#/definitions/ScriptSource"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/GalleryArtifactPublishingProfileBase"
+        }
+      ],
+      "required": [
+        "source"
+      ],
+      "description": "The publishing profile of a gallery image version."
+    },
+    "GalleryScriptVersionSafetyProfile": {
+      "type": "object",
+      "properties": {},
+      "allOf": [
+        {
+          "$ref": "#/definitions/GalleryArtifactSafetyProfileBase"
+        }
+      ],
+      "description": "The safety profile of the Gallery Script Version."
+    },
+    "ScriptSource": {
+      "type": "object",
+      "properties": {
+        "scriptLink": {
+          "type": "string",
+          "description": "Required. The link of the source script, it must be a readable storage blob with SAS URI or publicly accessible URI or managed identity enabled."
+        },
+        "parameters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GalleryScriptParameter"
+          },
+          "description": "Optional. Any input parameters that needs to passed to the script and are accessed within the script for its execution."
+        }
+      },
+      "required": [
+        "scriptLink"
+      ],
+      "description": "The source script from which the Script Version is going to be created."
+    },
+    "GalleryScriptParameter": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/GenericGalleryParameter"
+        }
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Specifies the type of the Gallery Script parameter. Possible values are: String, Int, Double, Boolean, Enum",
+          "enum": [
+            "String",
+            "Int",
+            "Double",
+            "Boolean",
+            "Enum"
+          ],
+          "x-ms-enum": {
+            "name": "GalleryScriptParameterType",
+            "modelAsString": false
+          }
+        },
+        "minValue": {
+          "type": "string",
+          "description": "The minimum value of parameter."
+        },
+        "maxValue": {
+          "type": "string",
+          "description": "The minimum value of parameter."
+        },
+        "enumValues": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": " A list of permissible values. Only applies to the Enum type."
+        }
+      },
+      "description": "The definition of a parameter that can be passed to a script of a Gallery Script Version."
+    },
+    "GenericGalleryParameter": {
+      "type": "object",
+      "properties": {
```

Comment:
>required [](http://example.com/codeflow?start=9&length=8)

standard guidance is to avoid booleans and prefer enums.

can this be enum ? 
RequirementStatus : Required, NotRequired ?

Reply:
we'd like to keep it bool at this time. GenericGalleryParameter class has been in use by other APIs since many years now. Thanks for the heads up for new APIs possibly.